apps/app/src/pages/Flow/config.tsx

```diff
@@ -1,53 +1,124 @@
-    icon: <LineChartAlt />,
-  },
-  dashboard: {
-    label: "Dashboard",
-    color: "cat2_80",
-    description: "Find here all the available dashboards.",
-    icon: <Cards />,
-  },
-} satisfies HvFlowProps<NodeGroup>["nodeGroups"];
+export const useNodeGroups = () => {
+  const { data: datasets } = useDatasets();
+
+  return useMemo<HvFlowNodeGroups>(
+    () => ({
+      dataset: {
+        label: "Dataset",
+        color: "cat3_80",
+        description: "Find here all the available datasets.",
+        icon: <Storage />,
+        items: datasets.map((ds) => ({
+          nodeType: "dataset",
+          label: ds.label,
+          data: {
+            endpoint: ds.url,
+            columns: ds.columns.map((column) => ({
+              id: column,
+              label: column,
+            })),
+          },
+        })),
+      },
+      visualization: {
+        label: "Visualization",
+        color: "cat1_80",
+        description: "Find here all the available visualizations.",
+        icon: <LineChartAlt />,
+        items: [
+          {
+            nodeType: "lineChart",
+            label: "Line Chart",
+            data: {
+              title: "",
+              measure: undefined,
+              groupBy: undefined,
+              splitBy: undefined,
+            },
+          },
+          {
+            nodeType: "barChart",
+            label: "Bar Chart",
+            data: {
+              title: "",
+              measure: undefined,
+              groupBy: undefined,
+              splitBy: undefined,
+            },
+          },
+          {
+            nodeType: "donutChart",
+            label: "Donut Chart",
+            data: {
+              title: "",
+              measure: undefined,
+              groupBy: undefined,
+            },
+          },
+          {
+            nodeType: "kpi",
+            label: "KPI",
+            data: {
+              title: "",
+              unit: "",
+              measure: undefined,
+              aggregation: undefined,
+            },
+          },
+          {
+            nodeType: "table",
+            label: "Table",
+            data: {
+              title: "",
+              columns: undefined,
+              measure: "EMEA",
+            },
+          },
+        ],
+      },
+      dashboard: {
+        label: "Dashboard",
+        color: "cat2_80",
+        description: "Find here all the available dashboards.",
+        icon: <Cards />,
+        items: [{ nodeType: "dashboard", label: "Dashboard" }],
+      },
+    }),
+    [datasets],
+  );
+};
 
 /** Node types */
-export const baseNodeTypes = {
+export const nodeTypes = {
   dashboard: Dashboard,
+  dataset: Dataset,
   lineChart: LineChart,
   barChart: BarChart,
   kpi: Kpi,
   donutChart: DonutChart,
   table: Table,
 } satisfies HvFlowProps["nodeTypes"];
 
-export type NodeType = keyof typeof baseNodeTypes;
+export type NodeType = keyof typeof nodeTypes;
 
 /** Initial Flow */
 export const nodes = [
```

Comment:
When we click on "Preview", the charts don't appear. We need to update line 160 in this file to `endpoint: "steelwheels1"` since the id changed.
Also, when we look at the nodes, only the node "Dataset" has a "subtitle". We should add the `nodeLabel` property to the other nodes too.